src/lib.rs

```diff
@@ -182,114 +194,131 @@ mod tests {
 
     #[test]
     fn find_solution_1_btc() {
-        let utxo_match = find_solution(ONE_BTC, COST_OF_CHANGE, &mut UTXO_POOL.clone()).unwrap();
-        let expected_bool_vec = vec![false, false, false, true];
-        assert_eq!(expected_bool_vec, utxo_match);
+        let utxo_match = select_coins_bnb(ONE_BTC, COST_OF_CHANGE, &mut UTXO_POOL.clone()).unwrap();
+        assert_eq!(1, utxo_match.len());
+        assert_eq!(ONE_BTC, utxo_match[0].get_value());
```

Comment:
Perhaps write these tests as a test function that takes a `target` and the expected input set as inputs. Then instead of having 11 functions, create a `map<target, input_set>` with 11 entries.

Reply:
Additionally, it seems to me that most of your tests are focused on exact matches of the target value. It would be good if you added 1) more tests that fall in the range between `lower_bound` and `upper_bound`, 2) a test that verifies that input sets matching `upper_bound` are accepted, 3) some tests with less round numbers, perhaps even multiple possible solutions, to test that you get the best, not the first solution, 4) some tests with bigger UTXO pools than 4 UTXOs.